core/domain/html_validation_service_test.py

```diff
@@ -1247,9 +1247,9 @@ def test_convert_svg_diagram_tags_to_image_tags(self):
             ),
             'expected_output': (
                 '<oppia-noninteractive-image '
-                'alt-with-value="&amp;quot;Image&amp;quot;" '
+                'alt-with-value=\'\"Image\"\' '
```

Comment:
Why are these changed ... isn't the escaping now incorrect? It should match what is seen in the HTML.

Reply:
The previous implementation of `convert_svg_diagram_tags_to_image_tags` didn't really work in practice -- it was not actually used in the codebase but rather only used in tests. I have tested this manually and it seems to preserve the text as-is in the HTML attributes:

Before migration:
```
"<oppia-noninteractive-svgdiagram alt-with-value=\"&amp;quot;Image&amp;quot;\" svg_filename-with-value=\"&amp;quot;img_20210728_005205_do73ocj8tp_height_350_width_450.svg&amp;quot;\"></oppia-noninteractive-svgdiagram>\n<p>abc</p>"
```

After migration:
```
"<oppia-noninteractive-image alt-with-value=\"&amp;quot;Image&amp;quot;\" caption-with-value=\"&amp;quot;&amp;quot;\" filepath-with-value=\"&amp;quot;img_20210728_005205_do73ocj8tp_height_350_width_450.svg&amp;quot;\"></oppia-noninteractive-image>\n<p>abc</p>"
```


Looking closely, it seems like the original tests assumed attibute values will be enclosed between "&quot;", which is not the case for tags in the datastore, it should instead be "&amp;quot;". I've fixed the test now.